lib/src/operance_data_column_width.dart

```diff
@@ -1,39 +1,34 @@
 // 🐦 Flutter imports:
-import 'package:flutter/widgets.dart';
+
 
 /// A class representing the width of a column in the OperanceDataTable.
 class OperanceDataColumnWidth {
   /// Creates an instance of [OperanceDataColumnWidth].
   ///
-  /// The [size] and [factor] parameters are optional.
-  /// If [size] is provided, it will be used as the column width.
+  /// The [size] parameter is optional and takes precedence over [factor].
+  /// If [size] is provided, it will be used as the fixed column width.
   /// If [size] is not provided, the width will be calculated based on the
-  /// [factor].
+  /// [factor], which defaults to 0.15 (15% of the available width).
   const OperanceDataColumnWidth({
     this.size,
     this.factor = 0.15,
-  });
+  }) : assert(factor > 0 && factor <= 1, 'Factor must be between 0 and 1');
 
   /// The factor to calculate the column width if [size] is not provided.
   ///
-  /// Defaults to `0.15`.
+  /// Must be a value between 0 and 1, representing a percentage of the
+  /// available width.
   final double factor;
 
-  /// The fixed size of the column.
-  ///
-  /// If provided, this value will be used as the column width.
+  /// If provided, this value will be used as the column width,
+  /// ignoring [factor].
   final double? size;
 
-  /// Returns the width of the column based on the [BuildContext].
```

Comment:
nice catch, initialy i was using the context.media.size.with to calculate it but ran in to issues when this widget is used in side other widgets with different sizes so had to change it to take in the widget provided from the LayoutBuilder widget but forgot to update the docs, thanks!